### PR TITLE
feat(backend): handle username always in lowercase

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
user 

### Description
This should make all usernames of new users into lowercase. Usernames are also searched in the DB as lowercase.

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

**Review by ErikMichelson — COMMENTED** [review]
Maybe it's nice to convert uppercase letters to lowercase in the register page while typing as well?

**Comment by davidmehren** [comment]
This has potential for unintuitive behavior, when the username is magically converted without a user noticing.

IMO, the proper way is to reject usernames with capital letters and return a corresponding error.

**Comment by mrdrogdrog** [comment]
This could also clash with e.g. LDAP.

**Comment by DerMolly** [comment]
> This has potential for unintuitive behavior, when the username is magically converted without a user noticing.

How so? It's common that `DerMolly`, `dermolly` and `DeRmOlLy` reference the same account, is it not?

I think it's a good idea to implement that the user input gets transformed on the frontend's side, so that the user directly sees that different capitalization don't work. This would affect:
- username on internal login
- user search in the permissions

> IMO, the proper way is to reject usernames with capital letters and return a corresponding error.

Given we transform all names in the frontend, we could respond with errors if people hold the api the wrong way.

**Comment by davidmehren** [comment]
According to my google-fu, LDAP `uid`s are case-insensitive by default. So *most* LDAPs should not be affected.

**Comment by mrdrogdrog** [comment]
We agreed on the following points:
  - Users can't be created if they have usernames with uppercase letters. The API should throw an error.
  - LDAP username will be lowercased on creation
  - The frontend should lowercase usernames automatically while typing

**Review by davidmehren — APPROVED** [review]

File: frontend/src/components/login-page/auth/fields/username-field.tsx [hunk 1]
```diff
@@ -8,19 +8,26 @@ import React from 'react'
 import { Form } from 'react-bootstrap'
 import { useTranslation } from 'react-i18next'
 
+export interface UsernameFieldProps extends AuthFieldProps {
+  value: string
+}
+
+//TODO: This should be replaced with the common username component
```
Comment (ErikMichelson) [inline frontend/src/components/login-page/auth/fields/username-field.tsx:15]: Is there an issue for this?

**Review by ErikMichelson — COMMENTED** [review]
Otherwise LGTM

File: frontend/src/components/login-page/auth/fields/username-field.tsx (hunk 1 above)
Comment (mrdrogdrog) [reply frontend/src/components/login-page/auth/fields/username-field.tsx]: https://github.com/hedgedoc/hedgedoc/issues/4128

**Review by ErikMichelson — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

**Pull request merged**